pyro/ops/integrator.py

```diff
@@ -34,7 +34,7 @@ def velocity_verlet(z, r, potential_fn, inverse_mass_matrix, step_size, num_step
                                                                         inverse_mass_matrix,
                                                                         step_size,
                                                                         z_grads)
-    return z_next, r_next, z_grads, potential_energy
```

Comment:
Could you share your model where this happens? At no point in HMC do we set `requires_grad = True`, except for when we take the gradient `_potential_grad` below, right after which this is set to False again. Are you by any chance using `pyro.param` in the model?

Reply:
Okay I see the issue. We are also returning `potential_energy` from `_potential_grad`. Could you instead make this change in `_potential_grad` below?

```python
return dict(zip(z_keys, grads)), potential_energy.detach()
```

Reply:
done